src/bots/filler.ts

```diff
@@ -897,6 +900,72 @@ export class FillerBot implements Bot {
 		return true;
 	}
 
+	protected selectSubsetOfMakers(nodeToFill: NodeToFill) {
+		let makerNodes = nodeToFill.makerNodes;
+		if (makerNodes.length >= MAX_MAKERS_PER_FILL) {
+			// Calculate the values for each item and sum them up
+			const values: BN[] = [];
+			const totalValue = ZERO;
+			for (const node of makerNodes) {
+				if (node.order) {
+					const value = node.order.baseAssetAmount.sub(
+						node.order.baseAssetAmountFilled
+					);
+					values.push(value);
+					totalValue.add(value);
+				}
+			}
+
+			// Calculate probabilities of skipping for each item
+			const probabilities: number[] = values.map((value) =>
+				value.mul(PERCENTAGE_PRECISION).div(totalValue).toNumber()
+			);
+
+			// Select MAX_MAKERS_PER_FILL unique items, biasing towards the ones in the front
+			const selectedItems: DLOBNode[] = [];
+			let tries = 0;
+
+			while (selectedItems.length < MAX_MAKERS_PER_FILL && tries < 10) {
```

Comment:
maybe do like `maxTries = MAX_MAKERS_PER_FILL * 1.5` so it's 1 less thing to worrya bout if we just bump MAX_MAKERS